checker/detail_logger_impl_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package checker
+
+import (
+	"testing"
+)
+
+func Test_logger_Info(t *testing.T) {
+	l := &logger{
+		logs: []CheckDetail{},
+	}
+	l.Info(&LogMessage{Text: "test"})
+	if len(l.logs) != 1 && l.logs[0].Type != DetailInfo {
+		t.Errorf("expected 1 log, got %d", len(l.logs))
+	}
+}
+
+func Test_logger_Warn(t *testing.T) {
+	l := &logger{
+		logs: []CheckDetail{},
+	}
+	l.Warn(&LogMessage{Text: "test"})
+	if len(l.logs) != 1 && l.logs[0].Type != DetailWarn {
+		t.Errorf("expected 1 log, got %d", len(l.logs))
+	}
+}
+
+func Test_logger_Flush(t *testing.T) {
+	l := &logger{
+		logs: []CheckDetail{},
+	}
+	l.Warn(&LogMessage{Text: "test"})
+	l.Flush()
```

Comment:
let's test the return value of flush here too.
perhaps that it has a length of one since we added a test message on the line above